src/three-components/Debugger.ts

```diff
@@ -0,0 +1,46 @@
+import {Mesh, OrthographicCamera, PlaneBufferGeometry, ShaderMaterial, Texture, WebGLRenderTarget} from 'three';
+
+import {Renderer} from './Renderer';
+
+export interface ModelViewerRendererDebugDetails {
+  renderer: Renderer;
+  THREE: {
+    ShaderMaterial: Constructor<ShaderMaterial>;
+    PlaneBufferGeometry: Constructor<PlaneBufferGeometry>;
+    OrthographicCamera: Constructor<OrthographicCamera>;
+    WebGLRenderTarget: Constructor<WebGLRenderTarget>;
+    Texture: Constructor<Texture>;
+    Mesh: Constructor<Mesh>;
+  };
+}
+
+/**
+ * This Debugger exposes internal details of the <model-viewer> rendering
+ * substructure so that external tools can more easily inspect and operate on
+ * them.
+ *
+ * It also activates shader debugging on the associated GL context. Shader
+ * debugging trades performance for useful error information, so it is not
+ * recommended to activate this unless needed.
+ */
+export class Debugger {
+  constructor(renderer: Renderer) {
+    renderer.renderer.debug = {checkShaderErrors: false};
+    Promise.resolve().then(() => {
+      self.dispatchEvent(new CustomEvent<ModelViewerRendererDebugDetails>(
+          'model-viewer-renderer-debug', {
+            detail: {
+              renderer,
+              THREE: {
+                ShaderMaterial,
```

Comment:
We should probably include `Scene` here, I think. Any others, @elalish ? My hope is to limit it to things that we already import for rendering, and are necessary to do essential debug rendering outside of the element.

Reply:
Agreed on Scene. Renderer already includes TextureUtils, so that's good. We might want to reevaluate what is private, for instance TextureUtils has a private member PMREMGenerator that it might be nice to access. Likewise perhaps with Renderer's private member arRenderer.

Reply:
I added `Scene`, and I added a read-only accessor for the `PMREMGenerator` in `TextureUtils`. Let's add `arRenderer` later if and when we need it.

Reply:
When you said Scene, I was actually thinking [$scene], since that owns things like the model and the shadow. The renderer has the set of all scenes, but it might be hard to tell which one relates to this element.

Reply:
Yah, currently there is no offered mechanism to distinguish the scenes. Originally, I added a special event to advertise each scene as it is registered with the `Renderer`, but I don't have a use case for it yet so I took it out.

I'm happy to add it back in if you think we should land it now.

Reply:
I don't have a strong opinion, but I like the idea of being able to access all of the textures. Many of those are in the Model's materials and the shadow lives in the ModelScene, so having hooks to that seems important.

Reply:
Added advertisement of scenes as they are registered/unregistered. Details in the PR description.